src/main/java/com/redhat/sast/api/service/BatchProcessingService.java

```diff
@@ -0,0 +1,220 @@
+package com.redhat.sast.api.service;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+import com.redhat.sast.api.enums.BatchStatus;
+import com.redhat.sast.api.enums.JobStatus;
+import com.redhat.sast.api.model.Job;
+import com.redhat.sast.api.model.JobBatch;
+import com.redhat.sast.api.repository.JobBatchRepository;
+import com.redhat.sast.api.util.input.CsvConverter;
+import com.redhat.sast.api.util.input.CsvJobParser;
+import com.redhat.sast.api.v1.dto.request.JobCreationDto;
+
+import jakarta.annotation.Nonnull;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * Shared service for common batch processing operations.
+ * Contains logic that is reused across different pipeline types.
+ */
+@ApplicationScoped
+@RequiredArgsConstructor
+@Slf4j
+public class BatchProcessingService {
```

Comment:
```suggestion
public class BatchHandler {
```

move it to a appropriate package as well.